### PR TITLE
[RSDK-1600] Add app image creation

### PR DESCRIPTION
**Pull request by kkufieta**
Ticket: https://viam.atlassian.net/browse/RSDK-1600

### CODE REVIEW

**Review by JohnN193 — APPROVED** [review]
Assuming this PR is for adding the appimage/canon implementation to the rplidar and the workflow/automation will be in a separate one, then LGTM

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,9 @@
 OS=$(shell uname)
 VERSION=v1.12.0
 CGO_LDFLAGS="-Lgen/third_party/rplidar_sdk-release-${VERSION}/sdk/output/${OS}/Release/"
+BUILD_CHANNEL?=local
```
Comment (tessavitabile) [inline Makefile:4]: [opt] You could probably remove the BUILD_CHANNEL and hardcode "latest" wherever BUILD_CHANNEL is used. Or you could leave this in place, in case we ever want to create a stable release channel.

File: etc/packaging/appimages/package_release.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Comment (tessavitabile) [inline etc/packaging/appimages/package_release.sh:1]: You can remove this script, since we are not tagging versions.

File: Makefile [hunk 3]
```diff
@@ -38,3 +46,12 @@ build-server: swig
 
 clean: clean-sdk
 	rm -rf bin gen/gen_wrap.cxx gen/gen.go
+
+appimage: build-module
+	cd etc/packaging/appimages && BUILD_CHANNEL=${BUILD_CHANNEL} appimage-builder --recipe rplidar-module-`uname -m`.yml
+	cd etc/packaging/appimages && ./package_release.sh
```
Comment (tessavitabile) [inline Makefile:52]: You can remove this line, since we are not tagging versions. (We only need to tag versions when using a stable release channel.)

File: etc/packaging/appimages/utils.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Comment (tessavitabile) [inline etc/packaging/appimages/utils.sh:1]: You can remove this script, since we are not tagging versions.

File: Makefile [hunk 5]
```diff
@@ -38,3 +46,12 @@ build-server: swig
 
 clean: clean-sdk
 	rm -rf bin gen/gen_wrap.cxx gen/gen.go
+
+appimage: build-module
```
Comment (tessavitabile) [inline Makefile:50]: Are you planning to do a separate PR to create the github workflow that will make the appimage (like this one: https://github.com/viamrobotics/slam/blob/main/.github/workflows/appimage.yml)? Sorry I missed this in the first round of review.

**Review by tessavitabile — CHANGES_REQUESTED** [review]

File: Makefile (hunk 5 above)
Comment (kkufieta) [reply Makefile:50]: Yes! I'll add you as a reviewer.

Comment (tessavitabile) [reply Makefile:50]: thanks!

**Review by tessavitabile — APPROVED** [review]

File: Makefile [hunk 6]
```diff
@@ -27,6 +27,13 @@ sdk:
 clean-sdk:
 	cd gen/third_party/rplidar_sdk-release-${VERSION}/sdk && $(MAKE) clean_sdk
 
+install-swig:
+ifeq ("Darwin", "$(shell uname -s)")
```
Comment (Otterverse) [inline Makefile:31]: Check for brew existing instead of mac-specific test. Non-debian linux can still be supported that way.

File: Makefile [hunk 7]
```diff
@@ -38,3 +45,16 @@ build-server: swig
...
+appimage: build-module
+	cd etc/packaging/appimages && appimage-builder --recipe rplidar-module-`uname -m`.yml
+	mkdir -p etc/packaging/appimages/deploy/
+	mv etc/packaging/appimages/*.AppImage* etc/packaging/appimages/deploy/
+	chmod a+rx etc/packaging/appimages/deploy/*.AppImage
+
+clean-appimage:
+	rm -rf etc/packaging/appimages/AppDir
+	rm -rf etc/packaging/appimages/appimage-build
+	rm -rf etc/packaging/appimages/deploy
```
Comment (Otterverse) [inline Makefile:58]: These should probably all be on line/command. Less chance of an error on one and then it exiting before cleaning the others.

File: README.md [hunk 8]
```diff
@@ -7,39 +7,35 @@ It has been tested on the following rplidars:
...
+      sudo curl -o /usr/local/bin/rplidar-module http://packages.viam.com/apps/rplidar/rplidar-module-latest-x86_64.AppImage
+      sudo chmod a+rx /usr/local/bin/rplidar-module
+      ```
+2. Run the [RDK](https://github.com/viamrobotics/rdk) web server using one of the example config files [modules/sample_osx.json](./module/sample_osx.json) or [modules/sample_linux.json](./module/sample_linux.json), depending on your operating system. 
+
+## Development
+
+### Installation
+1. `make`
+3. Dependencies for golang:
```
Comment (Otterverse) [inline README.md:33]: This isn't a new step (also, it'd be step 2 if it was.)

File: README.md [hunk 9]
```diff
@@ -7,39 +7,35 @@ It has been tested on the following rplidars:
...
+      ```bash
+      sudo curl -o /usr/local/bin/rplidar-module http://packages.viam.com/apps/rplidar/rplidar-module-latest-x86_64.AppImage
+      sudo chmod a+rx /usr/local/bin/rplidar-module
+      ```
+2. Run the [RDK](https://github.com/viamrobotics/rdk) web server using one of the example config files [modules/sample_osx.json](./module/sample_osx.json) or [modules/sample_linux.json](./module/sample_linux.json), depending on your operating system. 
+
+## Development
+
+### Installation
+1. `make`
```
Comment (Otterverse) [inline README.md:32]: Won't this JUST install the "all" target, which contains only swig?

**Review by Otterverse — CHANGES_REQUESTED** [review]
Looks mostly good. A couple minor changes requested. I'm assuming the canon files are just direct copies from RDK, so didn't analyze them in depth. If you did modify them, let me know.

**Review by Otterverse — APPROVED** [review]
LGTM
